tools/DeleteModules.ps1

```diff
@@ -0,0 +1,193 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidateSet("ModuleList", "AzureRMAndDependencies", "AzureAndDependencies", "NetCoreModules", "AzureStackAndDependencies")]
+    [string] $scope,
+    [Parameter(Mandatory = $false)]
+    [string[]] $listOfModules,
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("TestGallery", "PSGallery")]
+    [string] $repoName,
+    [Parameter(Mandatory = $false)]
+    [string] $nugetExe,
+    [Parameter(Mandatory = $false)]
+    [string] $apiKey
+)
+
+function Get-TargetModules
+{
+    [CmdletBinding()]
+    param
+    (
+      [string]$scope,
+      [string[]]$moduleList
+    )
+
+    if ($scope -eq "ModuleList") 
+    {
+        $targets = @()
+        $moduleList | ForEach-Object {
+            $targets += Find-Module -Name $_ -Repository $repoName
+        }
+
+        return $targets
+    }
+
+    else
+    {
+        $targets = @()
+        $query = ""
+        if ($scope -eq "AzureRMAndDependencies") {
+            $query = "AzureRM"
+        } elseif ($scope -eq "AzureAndDependencies") {
+            $query = "Azure"
+        } elseif ($scope -eq "NetCoreModules") {
+            $query = "AzureRM.Netcore"
+        } elseif ($scope -eq "AzureStackAndDependencies") {
+            $query = "AzureStack"
+        }
+
+        $azureRmAllVersions = Find-Module -Name $query -Repository PSGallery -AllVersions
+        $azureRmCurrent = $azureRmAllVersions[0]
+        $azureRmPrevious = $azureRmAllVersions[1]
+        $currentDependencies = Find-Module -Name $query -Repository $repoName -RequiredVersion $azureRmCurrent.Version -IncludeDependencies
+        $previousDependencies = Find-Module -Name $query -Repository $repoName -RequiredVersion $azureRmPrevious.Version -IncludeDependencies
```

Comment:
@maddieclayton these two calls to `Find-Module ... -IncludeDependencies` are _very_ time expensive. The same information we are getting with these calls (dependent module names and versions) is already contained in `$azureRmCurrent.Dependencies` and `$azureRmPrevious.Dependencies`.